lib/Conversion/HandshakeToHW/HandshakeToHW.cpp

```diff
@@ -674,12 +674,13 @@ ModuleDiscriminator::ModuleDiscriminator(Operation *op) {
         addUnsigned("FIFO_DEPTH", speculatorOp.getFifoDepth());
       })
       .Case<handshake::SpecSaveOp, handshake::SpecCommitOp,
-            handshake::SpeculatingBranchOp>([&](auto) {
+            handshake::SpeculatingBranchOp, handshake::NonSpecOp>([&](auto) {
         // No parameters needed for these operations
       })
-      .Case<handshake::SpecSaveCommitOp>([&](handshake::SpecSaveCommitOp saveCommitOp) {
-        addUnsigned("FIFO_DEPTH", saveCommitOp.getFifoDepth());
-      })
+      .Case<handshake::SpecSaveCommitOp>(
+          [&](handshake::SpecSaveCommitOp saveCommitOp) {
+            addUnsigned("FIFO_DEPTH", saveCommitOp.getFifoDepth());
+          })
```

Comment:
This is a formatter issue. The file has lines with the `) {}` pattern, which my formatter automatically changes to `){}`, so I disabled formatting for this file.

(This behavior is dependent on the version of clang-format)

 However, I noticed this line exceeds 80 characters.
 I’ve now applied the formatter to the speculation-related code to clean it up.